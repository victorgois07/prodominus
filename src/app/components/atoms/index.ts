export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
