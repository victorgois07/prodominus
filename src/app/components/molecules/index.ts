export * from "./CategoryFilter";
export * from "./ConfirmDialog";
export * from "./FormField";
export * from "./Modal";
export * from "./Pagination";
export * from "./SortFilter";
