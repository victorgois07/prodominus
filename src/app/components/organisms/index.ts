export * from "./FavoritesManagement";
export * from "./ProductCard";
export * from "./ProductFilters";
export * from "./ProductForm";
export * from "./ProductGrid";
export * from "./ProductManagement";
