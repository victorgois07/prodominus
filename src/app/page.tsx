"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useProductFilters } from "@/business/store/useProductFilters";
import {
  ConfirmDialog,
  Error,
  Loading,
  Pagination,
  ProductFilters,
  ProductManagement,
} from "./components";

export default function Home() {
  const { loading, error, totalPages } = useProducts();
  const { page, setPage } = useProductFilters();

  if (loading) return <Loading />;
  if (error) return <Error message={error} />;

  return (
    <main className="container mx-auto px-4 py-8">
      <h1 className="text-3xl font-bold text-gray-900 mb-8">Products</h1>

      <ProductFilters />

      <div className="mb-6 flex flex-col gap-4">
        <ProductManagement />
      </div>

      <Pagination page={page} totalPages={totalPages} onPage={setPage} />
      <ConfirmDialog />
    </main>
  );
}
