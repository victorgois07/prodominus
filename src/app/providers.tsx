"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { useState } from "react";

export function Providers({ children }: { children: React.ReactNode }) {
  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            staleTime: 1000 * 60 * 5,
            retry: 1,
            refetchOnWindowFocus: false,
            retryOnMount: true,
            refetchOnReconnect: true,
          },
        },
      })
  );

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
}
