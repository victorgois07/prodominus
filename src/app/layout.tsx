import type { Metadata } from "next";
import { Inter } from "next/font/google";
import { LoadingBackdrop } from "./components/atoms/LoadingBackdrop";
import "./globals.css";
import { Providers } from "./providers";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Prodominus - Product Management",
  description: "A modern product management system built with Next.js",
};

export default function RootLayout({
  children,
}: Readonly<{ children: React.ReactNode }>) {
  return (
    <html lang="en">
      <body className={`${inter.className} antialiased`}>
        <Providers>
          <header className="bg-white shadow-sm">
            <div className="container mx-auto px-4 py-4">
              <h1 className="text-2xl font-bold text-balance text-gray-900">
                Prodominus
              </h1>
            </div>
          </header>

          <main>{children}</main>

          <footer className="bg-gray-50 mt-8">
            <div className="container mx-auto px-4 py-6">
              <p className="text-center text-gray-600 text-sm">
                Â© {new Date().getFullYear()} Prodominus. All rights reserved.
              </p>
            </div>
          </footer>

          <LoadingBackdrop />
        </Providers>
      </body>
    </html>
  );
}
