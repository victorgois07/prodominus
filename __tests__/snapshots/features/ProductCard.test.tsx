import { ProductCard } from "@/app/components";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

describe("ProductCard Snapshot", () => {
  const mockProduct = {
    id: 1,
    title: "Test Product",
    price: 99.99,
    description: "Test Description",
    category: "test",
    image: "test.jpg",
  };

  const mockOnClick = vi.fn();

  it("should match snapshot", () => {
    const { container } = render(<ProductCard product={mockProduct} />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with missing image", () => {
    const productWithoutImage = { ...mockProduct, image: "" };
    const { container } = render(<ProductCard product={productWithoutImage} />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long title", () => {
    const productWithLongTitle = {
      ...mockProduct,
      title:
        "This is a very long product title that should be truncated in the UI",
    };
    const { container } = render(
      <ProductCard product={productWithLongTitle} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long description", () => {
    const productWithLongDescription = {
      ...mockProduct,
      description:
        "This is a very long product description that should be truncated in the UI. It contains multiple sentences and should be properly formatted.",
    };
    const { container } = render(
      <ProductCard product={productWithLongDescription} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with high price", () => {
    const productWithHighPrice = {
      ...mockProduct,
      price: 999999.99,
    };
    const { container } = render(
      <ProductCard product={productWithHighPrice} />
    );

    expect(container).toMatchSnapshot();
  });
});
