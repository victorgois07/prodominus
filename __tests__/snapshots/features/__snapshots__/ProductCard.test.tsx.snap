// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard Snapshot > should match snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
    role="article"
  >
    <div
      class="absolute top-2 left-2 z-10"
    >
      <button
        aria-label="Adicionar aos favoritos"
        class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
        type="button"
      >
        ü§ç
      </button>
    </div>
    <div
      class="absolute top-2 right-2 z-10 flex gap-2"
    />
    <div
      class="relative h-48 w-full"
    >
      <img
        alt="Test Product"
        class="object-contain p-4"
        src="test.jpg"
      />
    </div>
    <div
      class="p-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
      >
        Test Product
      </h3>
      <p
        class="text-gray-600 text-sm line-clamp-2 mb-4"
      >
        Test Description
      </p>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-xl font-bold text-green-600"
        >
          R$
          99,99
        </span>
        <span
          class="text-sm text-gray-500 capitalize"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshot > should match snapshot with high price 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
    role="article"
  >
    <div
      class="absolute top-2 left-2 z-10"
    >
      <button
        aria-label="Adicionar aos favoritos"
        class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
        type="button"
      >
        ü§ç
      </button>
    </div>
    <div
      class="absolute top-2 right-2 z-10 flex gap-2"
    />
    <div
      class="relative h-48 w-full"
    >
      <img
        alt="Test Product"
        class="object-contain p-4"
        src="test.jpg"
      />
    </div>
    <div
      class="p-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
      >
        Test Product
      </h3>
      <p
        class="text-gray-600 text-sm line-clamp-2 mb-4"
      >
        Test Description
      </p>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-xl font-bold text-green-600"
        >
          R$
          999.999,99
        </span>
        <span
          class="text-sm text-gray-500 capitalize"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshot > should match snapshot with long description 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
    role="article"
  >
    <div
      class="absolute top-2 left-2 z-10"
    >
      <button
        aria-label="Adicionar aos favoritos"
        class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
        type="button"
      >
        ü§ç
      </button>
    </div>
    <div
      class="absolute top-2 right-2 z-10 flex gap-2"
    />
    <div
      class="relative h-48 w-full"
    >
      <img
        alt="Test Product"
        class="object-contain p-4"
        src="test.jpg"
      />
    </div>
    <div
      class="p-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
      >
        Test Product
      </h3>
      <p
        class="text-gray-600 text-sm line-clamp-2 mb-4"
      >
        This is a very long product description that should be truncated in the UI. It contains multiple sentences and should be properly formatted.
      </p>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-xl font-bold text-green-600"
        >
          R$
          99,99
        </span>
        <span
          class="text-sm text-gray-500 capitalize"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshot > should match snapshot with long title 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
    role="article"
  >
    <div
      class="absolute top-2 left-2 z-10"
    >
      <button
        aria-label="Adicionar aos favoritos"
        class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
        type="button"
      >
        ü§ç
      </button>
    </div>
    <div
      class="absolute top-2 right-2 z-10 flex gap-2"
    />
    <div
      class="relative h-48 w-full"
    >
      <img
        alt="This is a very long product title that should be truncated in the UI"
        class="object-contain p-4"
        src="test.jpg"
      />
    </div>
    <div
      class="p-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
      >
        This is a very long product title that should be truncated in the UI
      </h3>
      <p
        class="text-gray-600 text-sm line-clamp-2 mb-4"
      >
        Test Description
      </p>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-xl font-bold text-green-600"
        >
          R$
          99,99
        </span>
        <span
          class="text-sm text-gray-500 capitalize"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshot > should match snapshot with missing image 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
    role="article"
  >
    <div
      class="absolute top-2 left-2 z-10"
    >
      <button
        aria-label="Adicionar aos favoritos"
        class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
        type="button"
      >
        ü§ç
      </button>
    </div>
    <div
      class="absolute top-2 right-2 z-10 flex gap-2"
    />
    <div
      class="relative h-48 w-full"
    >
      <img
        alt="Test Product"
        class="object-contain p-4"
        src="/images/placeholder.svg"
      />
    </div>
    <div
      class="p-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
      >
        Test Product
      </h3>
      <p
        class="text-gray-600 text-sm line-clamp-2 mb-4"
      >
        Test Description
      </p>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-xl font-bold text-green-600"
        >
          R$
          99,99
        </span>
        <span
          class="text-sm text-gray-500 capitalize"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;
